app/models/manageiq/providers/vmware/infra_manager/metrics_capture.rb

```diff
@@ -363,39 +363,24 @@ def perf_build_query_params(interval_by_mor, counter_info, start_time, end_time)
     params
   end
 
-  def perf_query(params, counter_info, interval_name)
+  def perf_query(params, counter_info)
     counters_by_mor = {}
     counter_values_by_mor_and_ts = {}
     return counter_values_by_mor_and_ts if params.blank?
 
     Benchmark.current_realtime[:num_vim_queries] = params.length
-    _log.debug { "Total item(s) to be requested: [#{params.length}], #{params.inspect}" }
 
-    query_size = concurrent_requests(interval_name)
-    vim_trips = 0
-    params.each_slice(query_size) do |query|
-      vim_trips += 1
+    _log.debug { "Starting request for [#{params.length}] item(s), #{params.inspect}" }
+    data, = Benchmark.realtime_block(:vim_execute_time) { @perf_vim_hist.queryPerfMulti(params) }
+    _log.debug { "Finished request for [#{params.length}] item(s)" }
 
-      _log.debug { "Starting request for [#{query.length}] item(s), #{query.inspect}" }
-      data, = Benchmark.realtime_block(:vim_execute_time) { @perf_vim_hist.queryPerfMulti(query) }
-      _log.debug { "Finished request for [#{query.length}] item(s)" }
-
-      Benchmark.realtime_block(:perf_processing) do
-        self.class.preprocess_data(data, counter_info, counters_by_mor, counter_values_by_mor_and_ts)
-      end
+    Benchmark.realtime_block(:perf_processing) do
+      self.class.preprocess_data(data, counter_info, counters_by_mor, counter_values_by_mor_and_ts)
     end
-    Benchmark.current_realtime[:num_vim_trips] = vim_trips
 
     return counters_by_mor, counter_values_by_mor_and_ts
   end
 
-  def concurrent_requests(interval_name)
-    query_size = super
-    # even when batching is turned off, still want to send multiple realtime records
-    query_size = 20 if query_size < 20 && interval_name == "realtime"
-    query_size
-  end
```

Comment:
removing this is what fixes the core build.

The calling code was expecting a `1` to be returned when batching is turned off, but since `1 < 20`, this was not returning `1` for `realtime`.